src/modals/EditModal.tsx

```diff
@@ -0,0 +1,152 @@
+import { useContext, useRef } from "react";
+import {
+  IonButton,
+  IonModal,
+  IonDatetime,
+  IonContent,
+  IonItem,
+} from "@ionic/react";
+import { useTranslation } from "react-i18next";
+import { format } from "date-fns";
+import "./EditModal.css";
+
+import type { Cycle } from "../data/ClassCycle";
+import { CyclesContext } from "../state/Context";
+import { useAverageLengthOfCycle } from "../state/CycleInformationHooks";
+
+interface PropsEditModal {
+  isOpen: boolean;
+  setIsOpen: (newIsOpen: boolean) => void;
+}
+
+const EditModal = (props: PropsEditModal) => {
+  const datetimeRef = useRef<null | HTMLIonDatetimeElement>(null);
+  const { t } = useTranslation();
+
+  const { cycles, updateCycles } = useContext(CyclesContext);
+  const averLengthOfCycle = useAverageLengthOfCycle();
+
+  const nowDate = new Date();
+  nowDate.setHours(0, 0, 0, 0);
+
+  const isActiveDates = (dateString: string) => {
+    if (cycles.length === 0) {
+      return true;
+    }
+    const date = new Date(dateString);
+    date.setHours(0, 0, 0, 0);
+
+    const lastCycleFinish: Date = new Date(cycles[0].startDate);
+    lastCycleFinish.setDate(lastCycleFinish.getDate() + averLengthOfCycle);
+    lastCycleFinish.setHours(0, 0, 0, 0);
+
+    return date.getTime() < lastCycleFinish.getTime();
+  };
+
+  function periodDays() {
+    const value: string[] = [];
+
+    for (const cycle of cycles) {
+      const startOfCycle = new Date(cycle.startDate);
+      startOfCycle.setHours(0, 0, 0, 0);
+      value.push(format(startOfCycle, "yyyy-MM-dd"));
+
+      for (let i = 1; i < cycle.periodLength; i++) {
+        startOfCycle.setDate(startOfCycle.getDate() + 1);
+        value.push(format(startOfCycle, "yyyy-MM-dd"));
+      }
+    }
+    return value;
+  }
+
+  return (
+    <IonModal
+      isOpen={props.isOpen}
+      backdropDismiss={false}
+    >
+      <IonContent
+        className="ion-padding"
+        color="basic"
+      >
+        <IonItem
+          color="basic"
+          lines="none"
+        ></IonItem>
+        <IonDatetime
+          class="edit-modal"
+          ref={datetimeRef}
+          color="white-basic"
+          presentation="date"
+          locale={t("locale")}
+          size="cover"
+          multiple
+          firstDayOfWeek={1}
+          value={periodDays()}
+          isDateEnabled={isActiveDates}
+        ></IonDatetime>
+
+        <IonItem
+          color="basic"
+          lines="none"
+        ></IonItem>
+
+        <IonButton
+          class="edit-buttons"
+          color="dark-basic"
+          fill="solid"
+          onClick={() => {
+            if (datetimeRef.current?.value) {
+              const periodDays = [datetimeRef.current.value].flat().sort();
```

Comment:
You can create `periodDays` const before `return` and use the same constant here and above in the `IonDateTime`'s `value` prop

Now it's looks strange that you calling `periodDays()` function above and here use `string` value of the same array and also `flat().sort()` it